extensions/blocks/repeat-visitor/components/edit.js

```diff
@@ -27,17 +27,18 @@ const RADIO_OPTIONS = [
 ];
 
 class RepeatVisitorEdit extends Component {
-	componentDidMount() {
-		this.props.setAttributes( { isThresholdValid: true } );
-	}
+	state = {
+		isThresholdValid: true,
+	};
 
 	setCriteria = criteria => this.props.setAttributes( { criteria } );
 	setThreshold = threshold => {
 		if ( /^\d+$/.test( threshold ) && +threshold > 0 ) {
```

Comment:
Could we shorten this to
```suggestion
		if ( parsetInt( threshold, 10 ) > 0 ) {
```

(see [MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/parseInt)) (And maybe even use the result from `parsetInt( threshold, 10 )` for `setAttributes`?

Reply:
This approach using `parseInt` is very similar to how our initial validation approach. However, we decided on going with the more rigorous regex check so that we could properly reject improper inputs like `"1.5"` and `"2+1"`.

I think we can safely replace typecasting `threshold` with a more explicit `parseInt` for the `setAttributes` line, however.

What do you think?

Reply:
Ah, okay. I guess I'm coming from a place of laziness here -- it seems like we're trying to perform a fairly boring task here (non-negative integer validation), so I was hoping we could use really simple tooling for that :slightly_smiling_face: 

I guess it's fine either way. Personally, I like `parseInt` better than the regex and unary `+` (and I don't care as much that it would allow quirky inputs like `"1.5"` and `"2+1"` -- it would transform them into something valid), but I don't care that much really, and I don't want to hold this PR nitpicking over minor things like that forever :sweat_smile: